apps/src/javalab/JavalabEditor.jsx

```diff
@@ -80,18 +81,21 @@ class JavalabEditor extends React.Component {
       // if there are changes to the editor, update redux.
       if (!tr.changes.empty && tr.newDoc) {
         this.props.setEditorText(tr.newDoc.toString());
+        this.onProjectChanged();
       }
     };
   };
 
   renameFileComplete(e) {
```

Comment:
General style suggestion: 
Once there gets to be more than one props or more than one state access, it's nice to pre-assign them to consts to shorten up the lines.
i.e.
`const {setFilename, filename} = this.props;`
Doing this also future-proofs against antipatterns such as naming a local variable after one of the props or state variables.